File: starter-bots/JavaBot/src/main/java/Models/GameObject.java
package Models;

import Enums.*;
import java.util.*;

public class GameObject {
  public UUID id;
  public Integer size;
  public Integer speed;
  public Integer currentHeading;
  public Position position;
  public ObjectTypes gameObjectType;

  public GameObject(UUID id, Integer size, Integer speed, Integer currentHeading, Position position, ObjectTypes gameObjectType) {
    this.id = id;
    this.size = size;
    this.speed = speed;
    this.currentHeading = currentHeading;
    this.position = position;
    this.gameObjectType = gameObjectType;
  }

  public UUID getId() {
    return id;
  }

  public void setId(UUID id) {
    this.id = id;
  }

  public int getSize() {
    return size;
  }

  public void setSize(int size) {
    this.size = size;
  }

  public int getSpeed() {
    return speed;
  }

  public void setSpeed(int speed) {
    this.speed = speed;
  }

  public Position getPosition() {
    return position;
  }

  public void setPosition(Position position) {
    this.position = position;
  }

  public ObjectTypes getGameObjectType() {
    return gameObjectType;
  }

  public void setGameObjectType(ObjectTypes gameObjectType) {
    this.gameObjectType = gameObjectType;
  }

  public static GameObject FromStateList(UUID id, List<Integer> stateList)
  {
    Position position = new Position(stateList.get(4), stateList.get(5));
    return new GameObject(id, stateList.get(0), stateList.get(1), stateList.get(2), position, ObjectTypes.valueOf(stateList.get(3)));
  }
}

File: starter-bots/NETCoreBot/NETCoreBot/Models/GameObject.cs
using NETCoreBot.Enums;
using System;
using System.Collections.Generic;

namespace NETCoreBot.Models
{
    public class GameObject
    {
        public Guid Id { get; set; }
        public int Size { get; set; }
        public int Speed { get; set; }
        public Position Position { get; set; }
        public int CurrentHeading { get; set; }
        public ObjectTypes GameObjectType { get; set; }
        
        public static GameObject FromStateList(Guid id, List<int> stateList)
        {
            return new GameObject
            {
                Id = id,
                Size = stateList[0],
                Speed = stateList[1],
                CurrentHeading = stateList[2],
                GameObjectType = (ObjectTypes) stateList[3],
                Position = new Position {X = stateList[4], Y = stateList[5]}
            };
        }
    }
}
